pkg/distributor/ha_tracker.go

```diff
@@ -602,7 +622,34 @@ func (h *defaultHaTracker) checkReplica(ctx context.Context, userID, cluster, re
 	return h.checkReplica(ctx, userID, cluster, replica, now)
 }
 
-func (h *defaultHaTracker) withinUpdateTimeout(now time.Time, receivedAt int64) bool {
+type defaultHaTrackerForUser struct {
+	*defaultHaTracker
+	userID              string
+	cfg                 HATrackerTimeoutsConfig
+	updateTimeoutJitter time.Duration
+}
+
+func (h *defaultHaTracker) forUser(userID string) defaultHaTrackerForUser {
+	uh := defaultHaTrackerForUser{
+		defaultHaTracker: h,
+		userID:           userID,
+		cfg: HATrackerTimeoutsConfig{
+			UpdateTimeout:          h.cfg.UpdateTimeout,
+			UpdateTimeoutJitterMax: h.cfg.UpdateTimeoutJitterMax,
+			FailoverTimeout:        h.cfg.FailoverTimeout,
+		},
+		updateTimeoutJitter: h.updateTimeoutJitter,
+	}
+	err := uh.cfg.overrideFromLimits(userID, h.limits)
```

Comment:
We're setting the `uh.cfg` from `h.cfg`, but I think that `uh.cfg.overrideFromLimits` overwrites all values, regardless whatever is set, isn't that true?

Reply:
See c8b56e7dd347c7fedad8de7ec195dd302a2afb76, the behavior should now be more clear.